### DISCUSSION TITLE
Looking to stream existing WebRTC stream to public.

### DISCUSSION THREAD

**Discussion by Gyrohammer** [owncast/owncast#3498] [Q&A]
Hello!

I've been searching around for a while now and not finding anyone with a similar use-case. I have a 3D printer and occasionally print things for friends, sometimes they like to check on the status of their print or whatnot. I currently have the printer running Mainsail with Crowsnest displaying a USB cam using webrtc. I wanted to use Owncast to simply host/pass-through the existing webrtc stream and then pass it through nginx to expose it. 

I feel like I'm misunderstanding the fundamentals on how owncast works. I know I could use obs but I'm running the printer off a headless rpi, so I have no UI to work with. I've tried with ffmpeg but alas the camera is busy since its already being used by crowsnest. Is there any way I could make this work? Am I barking up the wrong tree? 

Thanks!

**Comment by gabek** [comment]
In short, this doesn't actually have anything to do with Owncast.

I can't speak to the specifics of your setup, but as long as you can get RTMP in some way you can stream. It sounds like you'll need to find a way to ingest WebRTC and send out RTMP. You'll need to do this outside of Owncast, from the [broadcasting](https://owncast.online/docs/broadcasting/) side. Not from the Owncast side.

**Comment by Epy** [comment]
Hello
Wouldn't [Octoprint](https://octoprint.org/) answer to your need ?
It's web UI integrate a webcam stream if configured. I just don't know if you can set some registered visitors to watch without being able to modify configuration.

**Comment by martyjoy** [comment]
You shouldn't need Owncast for this, unless your goal is to either have the webcam image be a part of the owncast webpage or for it to be the actual stream you are serving.

Documentation from mainsail
https://crowsnest.mainsail.xyz/configuration/cam-section

Once you have your cams setup in mainsail to have an RTMP url all you would need to do is get NGINX to reverse proxy either a port or url to the cam stream as opposed to the Owncast port.
